backend/capellacollab/projects/capellamodels/modelsources/t4c/routes.py

```diff
@@ -1,71 +1,126 @@
 # SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
 # SPDX-License-Identifier: Apache-2.0
 
-
 import typing as t
 
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.exc import IntegrityError, NoResultFound
 from sqlalchemy.orm import Session
 
 import capellacollab.core.database as database
-import capellacollab.projects.capellamodels.modelsources.t4c.crud as database_projects
-from capellacollab.core.authentication.database import (
-    verify_admin,
-    verify_project_role,
-)
+from capellacollab.core.authentication.database import verify_admin
 from capellacollab.core.authentication.jwt_bearer import JWTBearer
-
-from . import models as schema_projects
+from capellacollab.projects.capellamodels.injectables import (
+    get_existing_capella_model,
+)
+from capellacollab.projects.capellamodels.models import DatabaseCapellaModel
+from capellacollab.projects.capellamodels.modelsources.t4c import crud
+from capellacollab.projects.capellamodels.modelsources.t4c.injectables import (
+    load_project_model,
+)
+from capellacollab.projects.capellamodels.modelsources.t4c.models import (
+    ResponseT4CModel,
+    SubmitT4CModel,
+    T4CRepositoryWithModels,
+)
+from capellacollab.projects.models import DatabaseProject
+from capellacollab.settings.modelsources.t4c.injectables import load_instance
+from capellacollab.settings.modelsources.t4c.repositories.models import (
+    DatabaseT4CRepository,
+)
+from capellacollab.settings.modelsources.t4c.repositories.routes import (
+    load_instance_repository,
+)
 
 router = APIRouter()
 
 
 @router.get(
     "/",
-    response_model=t.List[schema_projects.RepositoryProject],
+    response_model=list[ResponseT4CModel],
 )
-def get_t4c_model_for_model(
-    project: str,
+def list_t4c_models(
     db: Session = Depends(database.get_db),
-    token=Depends(JWTBearer()),
+    token: JWTBearer = Depends(JWTBearer()),
+    model=Depends(get_existing_capella_model),
+    repository: t.Optional[DatabaseT4CRepository] = Depends(
+        load_instance_repository
+    ),
 ):
-    verify_project_role(
-        project, allowed_roles=["manager", "administrator"], token=token, db=db
-    )
-    db_models = database_projects.get_all_t4c_models(db, project)
-    return db_models
+    verify_admin(token, db)
+    if not repository:
+        return crud.get_all_t4c_models(db, model)
+    return T4CRepositoryWithModels.from_orm(repository).models
```

Comment:
This branching is on purpose. This route accepts optional parameters `repository_id: t.Optional[int] =  None` and `t4c_instance_id: t.Optional[int] = None`. If they are set, then this route will return the T4C models of the repository. Else, it will return the T4C models of a Capella model.
I however removed the admin verification, and used the simple `return model.t4c_models` snippet.